raiden_contracts/contracts/services/UserDeposit.sol

```diff
@@ -0,0 +1,142 @@
+pragma solidity ^0.5.2;
+
+import "raiden/Token.sol";
+import "raiden/Utils.sol";
+
+contract UserDepositContract is Utils {
+    string constant public contract_version = "0.1.0";
+    uint constant public withdraw_delay = 100;  // time before withdraw is allowed in blocks
+
+    // Token to be used for the deposit
+    Token public token;
+
+    // Keep track of balances
+    mapping(address => uint256) public balances;
+    mapping(address => WithdrawPlan) public withdraw_plans;
+
+    /*
+     *  Structs
+     */
+    struct WithdrawPlan {
+        uint256 amount;
+        uint256 earliest_withdraw;  // earliest block at which withdraw is allowed
```

Comment:
Consider something like `withdraw_block` - to be more specific that is a `block` and not a `timestamp`, for example.